src/Entity/ItemId.php

```diff
@@ -23,8 +24,9 @@ class ItemId extends EntityId implements Int32EntityId {
 	 * @throws InvalidArgumentException
 	 */
 	public function __construct( $idSerialization ) {
-		$this->assertValidIdFormat( $idSerialization );
-		$this->serialization = strtoupper( $idSerialization );
+		parent::__construct( self::upcaseLocalId( $idSerialization ) );
```

Comment:
This method encodes knowledge about an implementation detail in the wrong class. The base class should not know anything about the fact that some subclasses do a magic "to upper case" conversion. Simply inline the `strtoupper` in the two subclasses.